datasets/za/pmg_legislators/crawler.py

```diff
@@ -0,0 +1,250 @@
+from typing import Dict, Optional, Set
+from followthemoney.helpers import post_summary
+from zavod import Context
+from zavod import helpers as h
+import re
+from urllib.parse import unquote
+from rigour.ids.wikidata import is_qid
+from zavod.logic.pep import categorise
+
+# add headers to the request
+
+PHONE_SPLITS = [",", "/", "(1)", "(2)", "(3)", "(4)", "(5)", "(6)", "(7)", "(8)"]
+PHONE_REMOVE = re.compile(r"(ex|ext|extension|fax|tel|\:|\-)", re.IGNORECASE)
+
+# List of regex patterns for positions of interest
+POSITIONS_OF_INTEREST = [
+    re.compile(r"National\s+Legislature", re.IGNORECASE),
+    re.compile(r"National\s+Assembly", re.IGNORECASE),
+    re.compile(r"National\s+Council of Provinces", re.IGNORECASE),
+    re.compile(r"Minister of", re.IGNORECASE),
+    re.compile(r"Provincial\s+Legislature", re.IGNORECASE),
+    re.compile(r"Member of the Provincial Legislature", re.IGNORECASE),
+    re.compile(r"Member of the Executive Committee", re.IGNORECASE),
+    re.compile(r"National\s+Executive", re.IGNORECASE),
+]
+
+
+def clean_emails(emails):
+    out = []
+    for email in h.multi_split(emails, ["/", ",", " or "]):
+        if email is None:
+            return
+        email = unquote(email)
+        email = email.strip()
+        email = email.rstrip(".")
+        out.append(email)
+    return out
+
+
+def clean_phones(phones):
+    out = []
+    for phone in h.multi_split(phones, PHONE_SPLITS):
+        phone = PHONE_REMOVE.sub("", phone)
+        out.append(phone)
+    return out
+
+
+def parse_person(context: Context, data: dict):
+    person_id = data.get("id")
+    if not person_id:
+        # context.log.error(f"Missing ID in person data: {data}")
+        return
```

Comment:
Let's use the slug from the identifier with scheme `pombola-slug` instead when available, but then fall back to this ID if it isn't.

Could you the also add sourceUrl with the pattern https://pa.org.za/person/{slug}/ if the slug is found?